lib/src/main/java/graphql/nadel/engine/blueprint/NadelFieldInstruction.kt

```diff
@@ -72,8 +73,16 @@ interface NadelGenericHydrationInstruction {
      * This can be the fields described in [NadelHydrationArgument.ValueSource.FieldResultValue.queryPathToField]
      * or [NadelBatchHydrationMatchStrategy.MatchObjectIdentifier.sourceId].
      */
+    @Deprecated("To be replaced by executableSourceFields")
     val sourceFields: List<NadelQueryPath>
 
+    /**
+     * The fields required to be queried on the source object in order to complete the hydration.
+     * This can be the fields described in [NadelHydrationArgument.ValueSource.FieldResultValue.queryPathToField]
+     * or [NadelBatchHydrationMatchStrategy.MatchObjectIdentifier.sourceId].
+     */
+    val executableSourceFields: List<ExecutableNormalizedField>
```

Comment:
We now build the entire `ExecutableNormalizedField` ahead of time instead of just computing the required `NadelQueryPath`